# Detect default component test cases that haven't been customized (`default-component-test`)

This rule detects when a component's test file only contains the default test cases generated by the component generator. This indicates that the developer has created a new component but hasn't written specific tests for its functionality.

## Rule Details

The component generator creates a default test file with three basic test cases that only verify basic React component requirements. While these tests are useful as a starting point, they don't validate the specific functionality of your component.

This rule will warn you when:
- You have only default test cases in your test file
- Individual test cases that match the default generated ones

Examples of **incorrect** code for this rule:

```js
describe.context.default('MyComponent', () => {
  it('should render without crashing', () => {
    // Default test implementation
  })

  it('should NOT render null', () => {
    // Default test implementation
  })

  it('should NOT extend classNames', () => {
    // Default test implementation
  })
})
```

Examples of **correct** code for this rule:

```js
describe.context.default('MyComponent', () => {
  it('should render without crashing', () => {
    // Default test implementation
  })

  it('should show error message when invalid input', () => {
    // Custom test validating specific component behavior
  })

  it('should trigger onSubmit when form is valid', () => {
    // Custom test validating specific component behavior
  })
})
```

### Options

This rule has no options.

## When Not To Use It

If you are working with non-component code or if you're satisfied with the basic React component validation that the default tests provide.

## Further Reading

- [React Testing Best Practices](https://reactjs.org/docs/testing.html)
- [Component Testing Guidelines](https://github.mpi-internal.com/scmspain/es-td-agreements/blob/master/30-Frontend/00-agreements)